src/Elmish.WPF/InternalTypes.fs

```diff
@@ -4,26 +4,39 @@ module internal Elmish.WPF.InternalTypes
 open System
 open System.Windows.Input
 
+module Option =
+  let apply a = Option.map (fun f -> f a)
+
 /// A command that optionally hooks into CommandManager.RequerySuggested to
 /// automatically trigger CanExecuteChanged whenever the CommandManager detects
 /// conditions that might change the output of canExecute. It's necessary to use
 /// this feature for command bindings where the CommandParameter is bound to
 /// another UI control (e.g. a ListView.SelectedItem).
-type Command(execute, canExecute, autoRequery) as this =
+type Command(execute: obj -> unit, canExecute: obj -> bool, autoRequery) as this =
+
+  let mutable _execute = Some execute
+
+  let mutable _canExecute = Some canExecute
 
-  let canExecuteChanged = Event<EventHandler,EventArgs>()
-  let handler = EventHandler(fun _ _ -> this.RaiseCanExecuteChanged())
+  let mutable _canExecuteChanged = new Event<EventHandler,EventArgs>()
```

Comment:
Why did you add `new`?

I thought that adding `new` here would cause a warning, but it didn't.  Through a warning, the compiler suggests that `new` should only be used to [construct classes](https://fsharpforfunandprofit.com/posts/classes/#constructing-and-using-a-class) that implement `IDisposable`, which [`Event<,>`](https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/control.event%5b'delegate,'args%5d-class-%5bfsharp%5d) does not.

Is the lack of warning for this code is a bug in the compiler?  If so, then this code would produce a warning when this bug is fixed.